lib/cocoapods-core/source/aggregate.rb

```diff
@@ -51,6 +51,7 @@ def all_sets
       #         name. Returns nil if no representative source found containing a pod with given name.
       #
       def representative_set(name)
+        warn '[DEPRECATION] `representative_set` is deprecated.  Please use `representative_set_for_search` instead.'
```

Comment:
please mark as deprecated in the API docs as well

Reply:
OK